include/oneapi/dpl/experimental/kt/internal/esimd_radix_sort_kernels.h

```diff
@@ -0,0 +1,706 @@
+                }
+                __grf_hist_summary_scan[0] = 0;
+                __grf_hist_summary_scan.template select<32, 1>(1) = __grf_hist_summary.template select<32, 1>(0);
+                _ONEDPL_PRAGMA_UNROLL
+                for (::std::uint32_t __i = 32; __i < __bin_count; __i += 32)
+                {
+                    __grf_hist_summary_scan.template select<32, 1>(__i + 1) =
+                        __grf_hist_summary.template select<32, 1>(__i) + __grf_hist_summary_scan[__i];
+                }
+                _ONEDPL_PRAGMA_UNROLL
+                for (::std::uint32_t __s = 0; __s < __bin_count; __s += 128)
+                {
+                    __dpl_esimd::__block_store_slm<::std::uint32_t, 64>(
+                        __bin_hist_slm_size + __s * sizeof(_HistT),
+                        __grf_hist_summary_scan.template select<128, 1>(__s).template bit_cast_view<::std::uint32_t>());
+                }
+            }
+            __dpl_esimd::__ns::barrier();
+            {
+                _ONEDPL_PRAGMA_UNROLL
+                for (::std::uint32_t __s = 0; __s < __bin_count; __s += 128)
+                {
+                    __bin_offset.template select<128, 1>(__s).template bit_cast_view<::std::uint32_t>() =
+                        __dpl_esimd::__block_load_slm<::std::uint32_t, 64>(__bin_hist_slm_size + __s * sizeof(_HistT));
+                }
+                if (__local_tid > 0)
+                {
+                    _ONEDPL_PRAGMA_UNROLL
+                    for (::std::uint32_t __s = 0; __s < __bin_count; __s += 128)
+                    {
+                        __dpl_esimd::__ns::simd<_HistT, 128> __group_local_sum;
+                        __group_local_sum.template bit_cast_view<::std::uint32_t>() =
+                            __dpl_esimd::__block_load_slm<::std::uint32_t, 64>((__local_tid - 1) * __hist_stride + __s * sizeof(_HistT));
+                        __bin_offset.template select<128, 1>(__s) += __group_local_sum;
+                    }
+                }
+            }
+            __dpl_esimd::__ns::barrier();
+        }
+
+        _ONEDPL_PRAGMA_UNROLL
+        for (::std::uint32_t __s = 0; __s < __data_per_work_item; __s += __data_per_step)
+        {
+            __dpl_esimd::__ns::simd<::std::uint16_t, __data_per_step> __bins_uw = __bins.template select<__data_per_step, 1>(__s);
+            __write_addr.template select<__data_per_step, 1>(__s) += __bin_offset.template iselect(__bins_uw);
+        }
+
+        if (__stage != __stage_count - 1)
+        {
+            _ONEDPL_PRAGMA_UNROLL
+            for (::std::uint32_t __s = 0; __s < __data_per_work_item; __s += __data_per_step)
+            {
+                __dpl_esimd::__vector_store<_KeyT, 1, __data_per_step>(
+                    __write_addr.template select<__data_per_step, 1>(__s) * sizeof(_KeyT),
+                    __keys.template select<__data_per_step, 1>(__s));
```

Comment:
I think it would be more clear if these overloads of `__vector_store` were renamed to include `SLM` in the name, similar to the function(s) they wrap.

Reply:
I have noted it in #1311 for addressing in the future